### ISSUE TITLE
unable to use copy and  paste options while using TextField within Navigator widget

### ISSUE THREAD

**Issue by sahilbadhan0** [flutter/flutter#41082]
Was empty

**Comment by escamoteur** [comment]
Please add the output of flutter doctor here.
Also on which platform does this happen?
Can you please add the code where you create the TextField.

**Comment by sahilbadhan0** [comment]
Sorry for late Response 

[✓] Flutter (Channel stable, v1.9.1+hotfix.2, on Mac OS X 10.14.3 18D109, locale en-GB)
 
[✓] Android toolchain - develop for Android devices (Android SDK version 28.0.3)
[✓] Xcode - develop for iOS and macOS (Xcode 10.1)
[✓] Android Studio (version 3.4)
[!] IntelliJ IDEA Community Edition (version 2019.1.3)
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
[✓] Connected device (1 available)

**Comment by sahilbadhan0** [comment]
import 'master_detail_view.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',

      home: MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);
  
  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    
    return Scaffold(
      body:
      MasterDetailView(),
    );
  }
}






///////Master Deatail
class MasterDetailView extends StatefulWidget {
 MasterDetailView();

  @override
  _MasterDetailViewState createState() => _MasterDetailViewState();
}

class _MasterDetailViewState extends State<MasterDetailView> {
  final double separatorWidth = 0.5;

  @override
  void initState() {
    // TODO: implement initState
    super.initState();
  }

  @override
  Widget build(BuildContext context) {
    return Row(
      mainAxisSize: MainAxisSize.max,
      children: <Widget>[
        Expanded(
          //Master
          flex: 3,
          child:
          new NewPage()
        ),
        Container(
          width: separatorWidth,
          height:MediaQuery.of(context).size.height,
          color: Colors.black,
        ),
        Expanded(
          flex: 7,
          child:  new Navigator(
            onGenerateRoute: DetailRouter.generateRoute,
            initialRoute: '/', // for default page
          ),

        )
      ],
    );
  }
}

class NewPage extends StatefulWidget {
  @override
  _NewPageState createState() => _NewPageState();
}

class _NewPageState extends State<NewPage> {
  @override
  Widget build(BuildContext context) {
    return Container(
     // child: TextField(),
    );
  }
}

//// Navigator
class DetailRouter {
  static Route<dynamic> generateRoute(RouteSettings setting) {
    print("Inside Detail route Builder");
    return new PageRouteBuilder(pageBuilder: (context, animation1, animation2) {
      switch (setting.name) {
        case '/':
          return NewPage();
        default:
          return new Container();
      }
    });
  }
}

class NewPage extends StatefulWidget {
  @override
  _NewPageState createState() => _NewPageState();
}

class _NewPageState extends State<NewPage> {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Container(
        child: Center(
            child: TextField(
              decoration: InputDecoration(
                hintText: "Enter text"
              ),
            )
        ),
      ),
    );
  }
}

**Comment by sahilbadhan0** [comment]
Actually 
I was creating app for I pad and i wanted to create MasterDetailView 
So i use Different navigators in my app for Master and detail and now Im having trouble in using textfield at Detail side

**Comment by escamoteur** [comment]
Could you please give it a try to switch to the master channel and see if this is still happening there?

For this do:

`flutter channel master`

`flutter upgrade`

**Comment by escamoteur** [comment]
Oh, and please tell me if this happens on Android and iOS or only on one platform.

**Comment by sahilbadhan0** [comment]
![front](https://user-images.githubusercontent.com/40284055/66413430-ff996800-ea14-11e9-8c97-5d781cd4ffd3.png)


In android I'm able to copy and paste as its on detail side but  unable to tap on Cut as its on master side .
Please check the screenshot of output and let me know if you have any suggestion

**Comment by escamoteur** [comment]
You mean if you tap on Cut nothing happens?
Could you try to provide us a small sample project that reproduces this problem?

**Comment by escamoteur** [comment]
I added the additonal triage because IMHO this could lead to a more generic problem in the framework

**Comment by Hixie** [comment]
@escamoteur thanks for flagging this. Framework seems like the right area to assign it to.

**Comment by iapicca** [comment]
The issue seems to occur with the latest master
```bash
[   +2 ms] Flutter run key commands.
[   +1 ms] r Hot reload. 🔥🔥🔥
[   +2 ms] R Hot restart.
[        ] h Repeat this help message.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] An Observatory debugger and profiler on Pixel 3a is available at: http://127.0.0.1:53682/aAQb4tJd0jY=/
[+25402 ms] Performing hot restart...
[   +6 ms] Refreshing active FlutterViews before restarting.
[  +32 ms] Scanned through 452 files in 12ms
[        ] Syncing files to device Pixel 3a...
[        ] Scanning asset files
[   +1 ms] <- reset
[        ] Compiling dart to kernel with 1 updated files
[   +1 ms] <- recompile package:master/main.dart e1674b08-fdf0-4c89-91aa-4778595aac79
[        ] package:master/main.dart
[        ] <- e1674b08-fdf0-4c89-91aa-4778595aac79
[ +112 ms] Updating files
[ +481 ms] DevFS: Sync finished
[        ] Syncing files to device Pixel 3a... (completed in 600ms)
[        ] Synced 18.6MB.
[   +1 ms] <- accept
[ +355 ms] Hot restart performed in 972ms.
[  +11 ms] Performing hot restart... (completed in 1,008ms)
[        ] Restarted application in 1,011ms.
[+1144 ms] I/flutter (13883): Inside Detail route Builder
[+4696 ms] W/IInputConnectionWrapper(13883): getExtractedText on inactive InputConnection
[   +1 ms] W/IInputConnectionWrapper(13883): getTextBeforeCursor on inactive InputConnection
[  +11 ms] W/IInputConnectionWrapper(13883): getExtractedText on inactive InputConnection
[   +1 ms] W/IInputConnectionWrapper(13883): getTextBeforeCursor on inactive InputConnection
[  +41 ms] W/IInputConnectionWrapper(13883): getExtractedText on inactive InputConnection
[   +1 ms] W/IInputConnectionWrapper(13883): getTextBeforeCursor on inactive InputConnection
[   +7 ms] E/SpannableStringBuilder(13883): SPAN_EXCLUSIVE_EXCLUSIVE spans cannot have a zero length
[ +325 ms] I/chatty  (13883): uid=10241(com.example.master) identical 2 lines
[        ] E/SpannableStringBuilder(13883): SPAN_EXCLUSIVE_EXCLUSIVE spans cannot have a zero length
[+3700 ms] E/SpannableStringBuilder(13883): SPAN_EXCLUSIVE_EXCLUSIVE spans cannot have a zero length
[+1811 ms] I/chatty  (13883): uid=10241(com.example.master) identical 2 lines
[   +1 ms] E/SpannableStringBuilder(13883): SPAN_EXCLUSIVE_EXCLUSIVE spans cannot have a zero length
[ +762 ms] I/flutter (13883): ══╡ EXCEPTION CAUGHT BY RENDERING LIBRARY ╞═════════════════════════════════════════════════════════
[   +1 ms] I/flutter (13883): The following assertion was thrown during layout:
[        ] I/flutter (13883): A RenderFlex overflowed by 80 pixels on the right.
[        ] I/flutter (13883):
[        ] I/flutter (13883): The relevant error-causing widget was:
[        ] I/flutter (13883):   MaterialApp file:///Users/nevercode/Desktop/projects/master/lib/main.dart:106:12
[        ] I/flutter (13883):
[        ] I/flutter (13883): The overflowing RenderFlex has an orientation of Axis.horizontal.
[        ] I/flutter (13883): The edge of the RenderFlex that is overflowing has been marked in the rendering with a yellow and
[        ] I/flutter (13883): black striped pattern. This is usually caused by the contents being too big for the RenderFlex.
[        ] I/flutter (13883): Consider applying a flex factor (e.g. using an Expanded widget) to force the children of the
[        ] I/flutter (13883): RenderFlex to fit within the available space instead of being sized to their natural size.
[        ] I/flutter (13883): This is considered an error condition because it indicates that there is content that cannot be
[        ] I/flutter (13883): seen. If the content is legitimately bigger than the available space, consider clipping it with a
[        ] I/flutter (13883): ClipRect widget before putting it in the flex, or using a scrollable container rather than a Flex,
[        ] I/flutter (13883): like a ListView.
[        ] I/flutter (13883): The specific RenderFlex in question is: RenderFlex#34df4 relayoutBoundary=up4 OVERFLOWING:
[        ] I/flutter (13883):   creator: Row ← ConstrainedBox ← Container ← DefaultTextStyle ← AnimatedDefaultTextStyle ←
[        ] I/flutter (13883):     _InkFeatures-[GlobalKey#15165 ink renderer] ← NotificationListener<LayoutChangedNotification> ←
[        ] I/flutter (13883):     PhysicalModel ← AnimatedPhysicalModel ← Material ← _TextSelectionToolbar ← CustomSingleChildLayout
[        ] I/flutter (13883):     ← ⋯
[        ] I/flutter (13883):   parentData: <none> (can use size)
[        ] I/flutter (13883):   constraints: BoxConstraints(0.0<=w<=117.7, h=44.0)
[        ] I/flutter (13883):   size: Size(117.7, 44.0)
[        ] I/flutter (13883):   direction: horizontal
[        ] I/flutter (13883):   mainAxisAlignment: start
[        ] I/flutter (13883):   mainAxisSize: min
[        ] I/flutter (13883):   crossAxisAlignment: center
[        ] I/flutter (13883):   textDirection: ltr
[        ] I/flutter (13883):   verticalDirection: down
[        ] I/flutter (13883): ◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤◢◤
[        ] I/flutter (13883): ════════════════════════════════════════════════════════════════════════════════════════════════════
```
```console
[✓] Flutter (Channel master, 1.18.0-9.0.pre.73, on Mac OS X 10.15.4 19E287, locale en-EE)
    • Flutter version 1.18.0-9.0.pre.73 at /Users/nevercode/development/flutter_master
    • Framework revision bd6ccb606a (15 hours ago), 2020-04-29 16:31:42 -0700
    • Engine revision 2db3276573
    • Dart version 2.9.0 (build 2.9.0-3.0.dev 726d3c7725)


[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/nevercode/Library/Android/sdk
    • Platform android-29, build-tools 29.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 11.3.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 11.3.1, Build version 11C504
    • CocoaPods version 1.9.0

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 3.6)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin version 45.1.1
    • Dart plugin version 192.7761
    • Java version OpenJDK Runtime Environment (build 1.8.0_212-release-1586-b4-5784211)

[✓] VS Code
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.9.1

[✓] Connected device (3 available)
    • Pixel 3a   • 965AY0WP5C • android-arm64  • Android 10 (API 29)
    • Web Server • web-server • web-javascript • Flutter Tools
    • Chrome     • chrome     • web-javascript • Google Chrome 81.0.4044.129

• No issues found!
```

[screenshot](https://user-images.githubusercontent.com/45637986/80725682-11a33b00-8b0c-11ea-9b12-34c954c93315.png)

**Comment by darshankawar** [comment]
Using the code sample mentioned here, https://github.com/flutter/flutter/issues/41082#issuecomment-537887488, I was properly able to use all the text selection toolbar options from master and detail view and vice versa.


https://user-images.githubusercontent.com/67046386/106599702-7306a780-657f-11eb-8c62-e0dea56bcf11.mp4

Closing this as fixed. If anyone disagrees, write in comments and I'll reopen it.

<details>
<summary>flutter doctor -v</summary>

```
[✓] Flutter (Channel stable, 1.22.6, on Mac OS X 10.15.4 19E2269 darwin-x64,
    locale en-IN)
    • Flutter version 1.22.6 at /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 9b2d32b605 (11 days ago), 2021-01-22 14:36:39 -0800
    • Engine revision 2f0af37152
    • Dart version 2.10.5

 
[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 12.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    • CocoaPods version 1.9.3

[!] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    ✗ Flutter plugin not installed; this adds Flutter specific functionality.
    ✗ Dart plugin not installed; this adds Dart specific functionality.
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

 
[✓] Connected device (2 available)            
    • SM A260G (mobile)            • 5200763ebcfa861f • android-arm • Android
      8.1.0 (API 27)
    • AOSP on IA Emulator (mobile) • emulator-5554    • android-x86 • Android 9
      (API 28) (emulator)

! Doctor found issues in 1 category.

[✓] Flutter (Channel master, 1.26.0-18.0.pre.121, on Mac OS X 10.15.4 19E2269
    darwin-x64, locale en-IN)
    • Flutter version 1.26.0-18.0.pre.121 at
      /Users/dhs/documents/Fluttersdk/flutter
    • Framework revision 225e408d4f (7 hours ago), 2021-02-01 22:26:03 -0500
    • Engine revision e59457b818
    • Dart version 2.12.0 (build 2.12.0-279.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.0)
    • Android SDK at /Users/dhs/Library/Android/sdk
    • Platform android-30, build-tools 30.0.0
    • Java binary at: /Applications/Android
      Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.3, Build version 12C33
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin
        code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see
      https://guides.cocoapods.org/using/getting-started.html#installation for
      instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b3-6915495)

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (4 available)
    • SM A260G (mobile)            • 5200763ebcfa861f • android-arm    • Android
      8.1.0 (API 27)
    • AOSP on IA Emulator (mobile) • emulator-5554    • android-x86    • Android
      9 (API 28) (emulator)
    • macOS (desktop)              • macos            • darwin-x64     • Mac OS
      X 10.15.4 19E2269 darwin-x64
    • Chrome (web)                 • chrome           • web-javascript • Google
      Chrome 88.0.4324.96

! Doctor found issues in 1 category.


```

</details>
